pkg/maps/lbmap/lbmap.go

```diff
@@ -561,3 +561,96 @@ func DeleteRevNATBPF(id loadbalancer.ServiceID, isIPv4 bool) error {
 	err := DeleteRevNat(revNATK)
 	return err
 }
+
+// DumpServiceMapsToUserspace dumps the contents of both the IPv6 and IPv4
+// service / loadbalancer BPF maps, and converts them to a SVCMap and slice of
+// LBSVC. IPv4 maps may not be dumped depending on if skipIPv4 is enabled. If
+// includeMasterBackend is true, the returned values will also include services
+// which correspond to "master" backend values in the BPF maps. Returns the
+// errors that occurred while dumping the maps.
+func DumpServiceMapsToUserspace(includeMasterBackend, skipIPv4 bool) (loadbalancer.SVCMap, []*loadbalancer.LBSVC, []error) {
+
+	newSVCMap := loadbalancer.SVCMap{}
+	newSVCList := []*loadbalancer.LBSVC{}
+	errors := make([]error, 0, 2)
+
+	parseSVCEntries := func(key bpf.MapKey, value bpf.MapValue) {
+		svcKey := key.(ServiceKey)
+		//It's the frontend service so we don't add this one
+		if svcKey.GetBackend() == 0 && !includeMasterBackend {
+			return
+		}
+		svcValue := value.(ServiceValue)
+
+		scopedLog := log.WithFields(logrus.Fields{
+			logfields.BPFMapKey:   svcKey,
+			logfields.BPFMapValue: svcValue,
+		})
+
+		scopedLog.Debug("parsing service mapping")
+		fe, be, err := serviceKeynValue2FEnBE(svcKey, svcValue)
+		if err != nil {
```

Comment:
Why isn't this error appended to the `errors` slice?

Reply:
Fixed in next upload.